bgpd/bgp_route.c

```diff
@@ -10806,6 +10826,17 @@ DEFUN(show_ip_bgp, show_ip_bgp_cmd,
 	int exact_match = 0;
 	struct bgp *bgp = NULL;
 	int idx = 0;
+	uint8_t show_flags = 0;
+
+	/* [<ipv4|ipv6> [all]] */
+	if (all) {
+		SET_FLAG(show_flags, BGP_SHOW_OPT_AFI_ALL);
+		if (argv_find(argv, argc, "ipv4", &idx))
+			SET_FLAG(show_flags, BGP_SHOW_OPT_AFI_IP);
+
+		if (argv_find(argv, argc, "ipv6", &idx))
+			SET_FLAG(show_flags, BGP_SHOW_OPT_AFI_IP6);
+	}
```

Comment:
"show ip bgp" is only meant for AFI = IPV4 only. So why is that we want to force it to show IPV6?

Reply:
show ip bgp ipv6 =>displays only ipv6 unicast entries.
show ip bgp ipv6 [all] => to display the ipv6 entries for all SAFIs (unicast & multicast)

Reply:
so here this flag is set whether to display only ipv6 entries or both ipv4 and ipv6 entries

Reply:
"show ip bgp" is deprecated and hence should be used to extend.

Reply:
This option is applicable for show bgp <ipv4|ipv6> also.